master/master.cfg

```diff
@@ -747,206 +770,204 @@ def freebsd_ami(abi, version):
+    ZFSEC2ENATestWorker(
+        name="CentOS-6-x86_64-testworker%s" % (str(i+1)),
         ami=centos6_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-centos7_x86_64_testslave = [
-    ZFSEC2TestSlave(
-        name="CentOS-7-x86_64-testslave%s" % (str(i+1)),
+centos7_x86_64_testworker = [
+    ZFSEC2TestWorker(
+        name="CentOS-7-x86_64-testworker%s" % (str(i+1)),
         ami=centos7_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-centos8_x86_64_testslave = [
-    ZFSEC2TestSlave(
-        name="CentOS-8-x86_64-testslave%s" % (str(i+1)),
+centos8_x86_64_testworker = [
+    ZFSEC2TestWorker(
+        name="CentOS-8-x86_64-testworker%s" % (str(i+1)),
         ami=centos8_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-debian10_x86_64_testslave = [
-    ZFSEC2TestSlave(
-        name="Debian-10-x86_64-testslave%s" % (str(i+1)),
+debian10_x86_64_testworker = [
+    ZFSEC2TestWorker(
+        name="Debian-10-x86_64-testworker%s" % (str(i+1)),
         ami=debian10_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-fedora32_x86_64_testslave = [
-    ZFSEC2TestSlave(
-        name="Fedora-32-x86_64-testslave%s" % (str(i+1)),
+fedora32_x86_64_testworker = [
+    ZFSEC2TestWorker(
+        name="Fedora-32-x86_64-testworker%s" % (str(i+1)),
         ami=fedora32_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-ubuntu18_x86_64_testslave = [
-    ZFSEC2TestSlave(
-        name="Ubuntu-18.04-x86_64-testslave%s" % (str(i+1)),
+ubuntu18_x86_64_testworker = [
+    ZFSEC2TestWorker(
+        name="Ubuntu-18-x86_64-testworker%s" % (str(i+1)),
         ami=ubuntu18_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-ubuntu20_x86_64_testslave = [
-    ZFSEC2TestSlave(
-        name="Ubuntu-20.04-x86_64-testslave%s" % (str(i+1)),
+ubuntu20_x86_64_testworker = [
+    ZFSEC2TestWorker(
+        name="Ubuntu-20-x86_64-testworker%s" % (str(i+1)),
         ami=ubuntu20_ami
-    ) for i in range(0, numtestslaves)
+    ) for i in range(0, numtestworkers)
 ]
 
-freebsd12_amd64_testslave = [
-    ZFSEC2TestSlave(
-        name="FreeBSD-stable/12-amd64-testslave%s" % (str(i+1)),
-        get_image=get_freebsd12_image
-    ) for i in range(0, numtestslaves)
+freebsd12_amd64_testworker = [
+    ZFSEC2TestWorker(
+        name="FreeBSD-stable_12-amd64-testworker%s" % (str(i+1)),
+        ami=get_freebsd12_image()
+    ) for i in range(0, numtestworkers)
 ]
 
-freebsd13_amd64_testslave = [
-    ZFSEC2TestSlave(
-        name="FreeBSD-head-amd64-testslave%s" % (str(i+1)),
-        get_image=get_freebsd13_image
-    ) for i in range(0, numtestslaves)
+freebsd13_amd64_testworker = [
+    ZFSEC2TestWorker(
+        name="FreeBSD-head-amd64-testworker%s" % (str(i+1)),
+        ami=get_freebsd13_image()
```

Comment:
We'll want to come up with a similar patch so this can be passed as a function again. The AMI needs to be determined when a worker is spawned rather than when the buildbot config is loaded.

FreeBSD is trying to get Amazon to provide us with an SSM parameter store, which would let us set a fixed parameter path as the AMI here and avoid this hack.